src/test/coins_tests.cpp

```diff
@@ -892,8 +885,13 @@ BOOST_AUTO_TEST_CASE(ccoins_write)
     for (const CAmount parent_value : {ABSENT, SPENT, VALUE1})
         for (const CAmount child_value : {ABSENT, SPENT, VALUE2})
             for (const char parent_flags : parent_value == ABSENT ? ABSENT_FLAGS : FLAGS)
-                for (const char child_flags : child_value == ABSENT ? ABSENT_FLAGS : CLEAN_FLAGS)
-                    CheckWriteCoins(parent_value, child_value, parent_value, parent_flags, child_flags, parent_flags);
+                for (const char child_flags : child_value == ABSENT ? ABSENT_FLAGS : CLEAN_FLAGS) {
```

Comment:
I might be completely off here, but in case of e.g. `CheckWriteCoins(ABSENT, SPENT , FAIL  , NO_ENTRY   , DIRTY|FRESH, NO_ENTRY   );` we're calling `.IsDirty()` which returns false, but `m_next` and `m_prev` are set because `CheckWriteCoins` calls them with different flags.

Can this happen during actual usage, i.e. shouldn't they only be part of the internal linked list when they're dirty?

It bothers me a lot how many of the simulated combinations don't always seem to reflect real usage - while comments warn us of situations that may not be covered with tests at all.
Would it maybe make sense to fix the reliability of the tests before this PR?